src/seedu/addressbook/commands/EditCommand.java

```diff
@@ -0,0 +1,92 @@
+package seedu.addressbook.commands;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.tag.Tag;
+
+/**
+ * Edits a person in the address book.
+ * Adds a person if no such person exists.
+ */
+public class EditCommand extends Command {
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edit a person in the address book. "
+            + "Contact details can be marked private by prepending 'p' to the prefix.\n"
+            + "Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...\n"
+            + "Example: " + COMMAND_WORD
+            + " John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2, #02-25 t/friends t/owesMoney";
+
+    public static final String MESSAGE_SUCCESS = "Person edited: %1$s";
+    public static final String MESSAGE_ALTERNATIVE = "No such person. "
+            + "A new person will be added\n"
+            + "Person added: %1$s";
+
+    private final Person toEdit;
+
+    /**
+     * Convenience constructor using raw values.
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
+    public EditCommand(String name,
+                      String phone, boolean isPhonePrivate,
+                      String email, boolean isEmailPrivate,
+                      String address, boolean isAddressPrivate,
+                      Set<String> tags) throws IllegalValueException {
+        final Set<Tag> tagSet = new HashSet<>();
```

Comment:
Why is this set `final`? I get that nothing else in the function should modify it but your code ensures that. And nothing outside the function has access to it since it is a local variable.